src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/actions/rad/RadAuth.java

```diff
@@ -0,0 +1,91 @@
+package network.radicle.jetbrains.radiclejetbrainsplugin.actions.rad;
+
+import com.intellij.execution.process.ProcessOutput;
+import com.intellij.openapi.application.ApplicationManager;
+import git4idea.repo.GitRepository;
+import network.radicle.jetbrains.radiclejetbrainsplugin.RadicleBundle;
+import network.radicle.jetbrains.radiclejetbrainsplugin.services.RadicleApplicationService;
+
+public class RadAuth implements RadAction {
+
+    private String notificationSuccessMsg;
+    private String errorMsg;
+    private String successMsg;
+    private String name;
+    private String passphrase;
+    private RadAuthAction action;
+
+    public RadAuth (String name, String passphrase, RadAuthAction action) {
+        this.name = name;
+        this.passphrase = passphrase;
+        this.action = action;
```

Comment:
since you already know the action, it would be best to prepare all necessary fields here (such as `successMsg`,etc), so that this class is closer to being immutable and easier to use/handle